crates/accelerate/src/target_transpiler/mod.rs

```diff
@@ -170,13 +233,13 @@ pub(crate) struct Target {
     #[pyo3(get, set)]
     pub concurrent_measurements: Option<Vec<Vec<PhysicalQubit>>>,
     gate_map: GateMap,
-    #[pyo3(get)]
-    _gate_name_map: IndexMap<String, TargetOperation, RandomState>,
-    global_operations: IndexMap<u32, HashSet<String>, RandomState>,
-    variable_class_operations: IndexSet<String, RandomState>,
-    qarg_gate_map: NullableIndexMap<Qargs, Option<HashSet<String>>>,
-    non_global_strict_basis: Option<Vec<String>>,
-    non_global_basis: Option<Vec<String>>,
+    gate_name_map: IndexMap<usize, TargetOperation, RandomState>,
```

Comment:
One small question here, is there a reason we the full width of a `usize`? It is natural because that's the index type for `IndexSet`, but I'm pretty confident we don't the full 64 bit (or 32 bit on those platform) address space. Realistically I feel like we could get away with a u8 because it seems unlikely for a target to support more than >255 gates at once. If we wanted to be a bit conservative we could use a u16, there is almost no possibility of a target having over 65k gates names it.

Reply:
I can imagine a world with >255 operation names, if we had something like multiple fractional 2q gates (heterogeneous arch) exposed both continuously and with various precisely calibrated points on the parametrisation.

Reply:
255 is more than it sounds like. I would be surprised if anyone spent the time to calibrate all those different variants even on a far future system. But I also agree with you that it is within the realm of plausibility, however `u16::MAX` is definitely sufficiently large though so I think using a u16 newtype here for the target string index would make a lot of sense.

Reply:
This makes sense. Although the default indexing unit for `IndexSet` is mainly `usize`, I see how it could help to use a smaller datatype to store these indices in.